### ISSUE TITLE
code-server through nginx proxies

### ISSUE THREAD

**Issue by jennasisis** [cdr/code-server#1214]
## Description
Hi everyone, I'm in a bit of a pickle.

Picture this, I have a website. For transparency I'll just use my actual domain, `akii.dev`.
I have said website running on an nginx instance on a raspberry pi.

I would like to add `code-server`, except I want it running on my local computer instead of my raspberry pi. Shouldn't be an issue, right?
```
server {
  listen 80;
  server_name code.akii.dev;

  location / {
    proxy_pass https://192.168.1.xxx:8080;
  }
}
```

And in response, Firefox says:
![image](https://user-images.githubusercontent.com/26691026/70394793-53e5a780-19c6-11ea-998a-36128b3b2c69.png)

My predicament is as follows: **.dev domains require https.** When I start code server with `--cert` so it has its own self-signed cert, the HSTS issue appears. When i start code server *without* `--cert` the HSTS issue appears.

Now normally this wouldn't be an issue, however I can't seem to get certbot working for some odd reason. Whenever I try, it gives me this:
![image](https://user-images.githubusercontent.com/26691026/70394857-061d6f00-19c7-11ea-812f-eff144446336.png)

I am in multiple layers of "what the hell do I do now". Any help would be greatly appreciated.

# Possible related issues:
https://github.com/cdr/code-server/issues/80

**Comment by jennasisis** [comment]
Ok! Not sure how I did it, but I used a cert that I generated earlier.. I'm not sure when I did it and how I did but hey I did it.

I changed the port to listen on to 443 and added in the certs, then I used the proxy info from doc/quickstart.md and it all worked.

```
server {
  listen 443;
  listen [::]:443;
  server_name code.akii.dev;

  location / {
    proxy_pass http://192.168.1.208:8080;
    proxy_set_header Host $host;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection upgrade;
    proxy_set_header Accept-Encoding gzip;
  }

  ssl on;
  ssl_certificate /cert/path;
  ssl_certificate_key /key/path;
}
```

**Comment by jennasisis** [comment]
Ok! Not sure how I did it, but I used a cert that I generated earlier.. I'm not sure when I did it and how I did but hey I did it.

I changed the port to listen on to 443 and added in the certs, then I used the proxy info from doc/quickstart.md and it all worked.

server {
  listen 443;
  listen [::]:443;
  server_name code.akii.dev;

  location / {
    proxy_pass http://192.168.1.xxx:8080;
    proxy_set_header Host $host;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection upgrade;
    proxy_set_header Accept-Encoding gzip;
  }

  ssl on;
  ssl_certificate /cert/path;
  ssl_certificate_key /key/path;
}
